server/src/test/java/org/elasticsearch/index/engine/CombinedDeletionPolicyTests.java

```diff
@@ -315,44 +325,120 @@ synchronized boolean releaseCommit(IndexCommit indexCommit) {
         combinedDeletionPolicy.onInit(List.of(commit0));
 
         assertThat(acquiredCommits, contains(commit0));
         assertThat(deletedCommits, hasSize(0));
+        assertThat(newCommitFiles, contains("segments_0"));
 
-        final IndexCommit commit1 = mockIndexCommit(10L, 10L, translogUUID);
+        final IndexCommit commit1 = mockIndexCommit(10L, 10L, translogUUID, Set.of("_0.cfe", "_0.si", "_0.cfs", "segments_1"));
         combinedDeletionPolicy.onCommit(List.of(commit0, commit1));
 
         assertThat(acquiredCommits, contains(commit0, commit1));
         assertThat(deletedCommits, hasSize(0));
+        assertThat(newCommitFiles, containsInAnyOrder(equalTo("_0.cfe"), equalTo("_0.si"), equalTo("_0.cfs"), equalTo("segments_1")));
 
         globalCheckpoint.set(10L);
-        final IndexCommit commit2 = mockIndexCommit(20L, 20L, translogUUID);
+        final IndexCommit commit2 = mockIndexCommit(
+            20L,
+            20L,
+            translogUUID,
+            Set.of("_1.cfs", "_0.cfe", "_0.si", "_1.cfe", "_1.si", "_0.cfs", "segments_2")
+        );
         combinedDeletionPolicy.onCommit(List.of(commit0, commit1, commit2));
 
         assertThat(acquiredCommits, contains(commit0, commit1, commit2));
         assertThat(deletedCommits, hasSize(0));
+        assertThat(newCommitFiles, containsInAnyOrder(equalTo("_1.cfe"), equalTo("_1.si"), equalTo("_1.cfs"), equalTo("segments_2")));
 
         boolean maybeCleanUpCommits = combinedDeletionPolicy.releaseCommit(commit0);
         assertThat(maybeCleanUpCommits, equalTo(true));
 
         globalCheckpoint.set(20L);
-        final IndexCommit commit3 = mockIndexCommit(30L, 30L, translogUUID);
+        final IndexCommit commit3 = mockIndexCommit(
+            30L,
+            30L,
+            translogUUID,
+            Set.of(
+                "_3.fdx",
+                "_3_Lucene90_0.tip",
+                "_3_Lucene90_0.dvm",
+                "_3.si",
+                "_3_0.tmd",
+                "_3_0.tim",
+                "_3_ES85BloomFilter_0.bfi",
+                "_3_0.pos",
+                "_3_ES85BloomFilter_0.bfm",
+                "_3_0.tip",
+                "_3_Lucene90_0.doc",
+                "_3.nvd",
+                "_3.nvm",
+                "_3.fnm",
+                "_3_0.doc",
+                "segments_3",
+                "_3.kdd",
+                "_3_Lucene90_0.tmd",
+                "_3.fdm",
+                "_3.kdi",
+                "_3_Lucene90_0.dvd",
+                "_3_Lucene90_0.pos",
+                "_3.kdm",
+                "_3.fdt",
+                "_3_Lucene90_0.tim"
+            )
+        );
         combinedDeletionPolicy.onCommit(List.of(commit0, commit1, commit2, commit3));
 
         assertThat(acquiredCommits, contains(commit1, commit2, commit3));
         assertThat(deletedCommits, contains(commit0));
+        assertThat(newCommitFiles, hasItem("segments_3"));
+        assertThat(newCommitFiles, hasSize(25));
```

Comment:
Why not `containsInAnyOrder` here ?
I feel like it'd be best to have the hasItem, hasSize and containsInAnyOrder statements in all places above and below.

Reply:
Sure, I pushed 0d8c029c55a8bd842aeb943e3e2c3a5279d6c507